data/settings/src/main/java/com/google/jetpackcamera/settings/model/SupportedStabilizationMode.kt

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright (C) 2023 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.jetpackcamera.settings.model
+
+/** Enum class representing the device's supported video stabilization configurations. */
+enum class SupportedStabilizationMode {
+    /** Device supports both Preview and Video stabilization. */
+    FULL,
+
+    /** Device supports only Video stabilization.*/
+    VIDEO_ONLY,
+
+    /** Device supports only Preview stabilization.*/
+    PREVIEW_ONLY,
+
+    /** Device doesn't support any stabilization.*/
+    UNSUPPORTED
```

Comment:
This seems like it will become difficult to maintain as separate states if we were to add one more. For instance, if we added another stabilization mode called `SUPER_STABLE`, then we'll need 4 more enums: `SUPER_STABLE_ONLY`, `SUPER_STABLE_AND_VIDEO_ONLY`, `SUPER_STABLE_AND_PREVIEW_ONLY` `PREVIEW_AND_VIDEO_ONLY`. If you add _even more_ modes, the number of enums needed will start to become untenable. 

How about we just make `SupportedStabilizationMode` have `ON` and `HIGH_QUALITY`. We can then pass it around as `val supportedStabilizationModes: List<SupportedStabilizationMode>`. Then in our settings, we disable the entire stabilization setting if `supportedStabilizationModes.isEmpty()`, and for each of the individual choices, `enabled == supportedStabilizationModes.contains(SupportedStabilizationMode.***)`. We could still leave the protos as individual settings, but in LocalSettingsRepository, we just have:

```
private fun getSupportedStabilization(
        previewSupport: Boolean,
        videoSupport: Boolean
    ): List<SupportedStabilizationMode> {
  return buildList {
    if (previewSupport) add(SupportedStabilizationMode.ON)
    if (videoSupport) add(SupportedStabilizationMode.HIGH_QUALITY)
  }
}
```